app/Http/Middleware/VerifyPrivilegedUser.php

```diff
@@ -24,6 +24,6 @@ class VerifyPrivilegedUser extends VerifyUser
 {
     public function requiresVerification($request)
     {
-        return $this->auth->user() === null || $this->auth->user()->isPrivileged();
+        return session()->get('requires_verification') ?? optional(auth()->user())->isPrivileged();
```

Comment:
need some way of invalidating `requires_verification` on privilege change

Reply:
Not sure how to cleanly handle that on notification server side.

Probably easier to just kick out the user's session.

Reply:
if it's for the notification server, the web side shouldn't need to rely on the session value?

Reply:
Both should rely on same thing...?

Reply:
one checks per-request and the other is a persistent connection, I don't think it should be considered the same?

Reply:
More like per session...

Reply:
moving a logged in user to a privileged group currently triggers a verification on the next request, but this change makes it no longer trigger the verification

Reply:
Not sure this is a real issue, to be honest.

Reply:
Either:
- Privilege changes should force logout a user,
- the session value should be invalidated,
- or the session value should not be used